common-pg/src/main/scala/com/socrata/pg/soql/SoQLAnalysisSqlizer.scala

```diff
@@ -454,17 +454,30 @@ trait SoQLAnalysisSqlizer {
     // * select ... from single_row => select ...
     // * select ... from single_row join other_table on true .. => select ... from other_table ...
 
-    val (from, finalJoinPhrase, finalSetParamsJoin) =
-      if ((prevAna.isEmpty && fromTableName.isEmpty) || (fromTableName.map(_.name) == Some(TableName.SingleRow) && joinPhrase.isEmpty)) ("", joinPhrase, setParamsJoin)
-      else if (fromTableName.map(_.name) == Some(TableName.SingleRow) && joins.head.typ == InnerJoinType && joins.head.on == BooleanLiteral(true, SoQLBoolean)(joins.head.on.position)) {
-        (s" FROM ${joinPhrase.head.tableSql}", joinPhrase.tail, setParamsJoin.tail)
-      } else (s" FROM ${tableNamesWithThis(tableName.copy(alias = None))}" + tableName.alias.map(a => s" as ${a}").getOrElse(""), joinPhrase, setParamsJoin)
+    val (from, fromParams, joinPhrase, setParamsJoin) =
+      if ((prevAna.isEmpty && fromTableName.isEmpty) || (fromTableName.map(_.name) == Some(TableName.SingleRow) && joinPhrasesAndParams.isEmpty)) {
+        val (joinPhrases, joinSetParamses) = joinPhrasesAndParams.unzip
+        ("", Nil,
+         joinPhrases, joinSetParamses.flatten)
+      } else if (fromTableName.map(_.name) == Some(TableName.SingleRow) && joins.head.typ == InnerJoinType && joins.head.on == BooleanLiteral(true, SoQLBoolean)(joins.head.on.position)) {
+        // we're joining single_row to a thing on true, so pop that
```

Comment:
I appreciate the comment but I gotta admit, I feel like I understand enough of this to go "yes, I think I see how the code is doing this" but not at all why

Reply:
Just to make sure we're 100% on the same page, what I'm doing here is rewriting
```
select * from single_row join (something) as x on true ...
```
into
```
select * from (something) as x ...
```
leaving any further joins behind if there were any.  So I'm popping the parameters from the first join off and associating it with the from fragment, but there's the one little wrinkle that the param list will contain a parameter for the `true` in the join condition.  Since I know what that is we can just `dropRight` it off the param list (keeping any associated with the `(something)` part).

As for the why, it's to hopefully make the postgresql query optimizer as happy has possible.  Its statistics will know that `single_row` is a small table, but what's even better than joining to a table that's known to be small is not joining to it at all, and since `single_row` exists to support UDFs, we also know that this sort of join-on-true will be a common pattern (since we generate them during the UDF expansion)